### PR TITLE
Improve Settings pages UI

### PR DESCRIPTION
**Pull request by DekelR**
## Description: 
Improve Settings pages UI.

## Type of change:
- [ ] Bug fix
- [ ] New feature
- [X] Fix/new infrastructure

## Background context of the changes:
Improve UI.

## The ticket this PR closes:
[Put a link to the ticket on Jira we're closing using this pull request.
](https://ofirlindekel.atlassian.net/browse/NOTIFY-165?atlOrigin=eyJpIjoiYzE0ZTYyYmVlY2U0NDhhYmJkMWE3YjQzOTk0MjdmZmEiLCJwIjoiaiJ9)

## Where should the reviewer start?
All four settings pages.

## Insert screenshots / video of test results (for all relevant platforms)

image
image
image
image
image

### CODE REVIEW

File: Notify/Notify/Notify/Views/LocationSettingsPage.xaml [hunk 1]
```diff
@@ -12,86 +12,118 @@
...
-                    <materialDesignControls1:MaterialPicker.ItemsSource>
-                        <Binding Path="DropBoxOptions" />
-                    </materialDesignControls1:MaterialPicker.ItemsSource>
-                </materialDesignControls1:MaterialPicker>
+                    HeightRequest="20"
+                    Orientation="Horizontal"
+                    Spacing="10">
+                    <CheckBox Color="Black" IsChecked="{Binding IsUseCurrentLocation, Mode=TwoWay}" Margin="30,0,0,0" />
+                    <Label
+                        Text="USE CURRENT LOCATION"
```
Comment (LinMoskovitch) [inline Notify/Notify/Notify/Views/LocationSettingsPage.xaml:59]: Use Current Location

File: Notify/Notify/Notify/Views/LocationSettingsPage.xaml [hunk 2]
```diff
@@ -12,86 +12,118 @@
...
+                    Grid.Row="6"
+                    Margin="50,40,50,20"
                     BackgroundColor="{AppThemeBinding Light={StaticResource LightButtonColor},
                                                       Dark={StaticResource DarkButtonColor}}"
                     Command="{Binding GetAddressSuggestionsCommand}"
+                    IsVisible="{Binding m_IsUseOtherLocation}"
                     CornerRadius="30"
-                    Grid.Row="4"
-                    Margin="5,40,5,0"
                     Text="GET ADDRESS SUGGESTIONS"
```
Comment (LinMoskovitch) [inline Notify/Notify/Notify/Views/LocationSettingsPage.xaml:123]: Get Address Suggestion

**Review by LinMoskovitch — CHANGES_REQUESTED** [review]

**Comment by LinMoskovitch** [comment]
Great job !!
Regardless of what I have already written:

1. Try to put the Title of each page, e.g. Location Settings, in the same horizontal line as the back button.
Take a look at what I did on the Create Notification page, I managed to put them in the same line.
Try to do that on each of the 3 pages.

2. In the labels above the entries, there is a little inconsistency in the naming. For example: Select destination and Enter Address. I think it would be better to use "title case" for each of the labels

3. Try to limit the size of the UPDATE LOCATION button because it spreads on the entire screen from left to right.

Other than that it looks very good.

File: Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs [hunk 3]
```diff
@@ -172,6 +170,36 @@ private async void onBackButtonClicked()
 
         #endregion
 
+        #region Use_Current_Location
+
+        private bool m_IsUseCurrentLocation;
```
[the remark this answers is not in the document]
Comment (DekelR) [reply Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs:175]: I disagree - bool members need to start with "is___"

File: Notify/Notify/Notify/Views/LocationSettingsPage.xaml (hunk 2 above)
Comment (DekelR) [reply Notify/Notify/Notify/Views/LocationSettingsPage.xaml:123]: Done

File: Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs (hunk 3 above)
[the remark this answers is not in the document]
Comment (LinMoskovitch) [reply Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs:175]: I know but there is no sense in IsUse...

File: Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs [hunk 4]
```diff
@@ -172,6 +170,36 @@ private async void onBackButtonClicked()
...
+            get => m_IsUseCurrentLocation;
+            set
+            {
+                m_IsUseCurrentLocation = value;
+                OnPropertyChanged(nameof(IsUseCurrentLocation));
+
+                if (m_IsUseCurrentLocation)
+                {
+                    m_TmpLongitude = Longitude;
+                    m_TmpLatitude = Latitude;
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs:189]: Use a different name than m_TmpLongitude / m_TmpLatitude.
This is the current longitude and latitude, so m_CurrentLongitude / m_CurrentLatitude is a better option here.

File: Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs [hunk 5]
```diff
@@ -172,6 +170,36 @@ private async void onBackButtonClicked()
...
+                    getCurrentLocationButtonClicked();
+                }
+                else
+                {
+                    Longitude = m_TmpLongitude;
+                    Latitude = m_TmpLatitude;
+                }
+            }
+        }
+        public bool m_IsUseOtherLocation => !m_IsUseCurrentLocation;
```
Comment (OfirMatasas) [inline Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs:199]: There's a convertor for this kind of things.
You can ask @LinMoskovitch for how you can do it.

**Review by OfirMatasas — DISMISSED** [review]
Looks good overall.
Just make sure to fix the things I mentioned.

File: Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs (hunk 4 above)
Comment (DekelR) [reply Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs:189]: It's not - I'm saving here the coordinates of the address that the user picked, so if he unchecks the 'use current location' checkbox I will still have the old coordinates.

Comment (OfirMatasas) [reply Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs:189]: Then use another name.
Something like suggestedAddressLatitude

Comment (DekelR) [reply Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs:189]: m_SelectedAddressLongitude
m_SelectedAddressLatitude

File: Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs (hunk 5 above)
Comment (DekelR) [reply Notify/Notify/Notify/ViewModels/LocationSettingsPageViewModel.cs:201]: When using the converter the 'IsEnabled' is not working properly - the fields only change color to gray but they are still functional and responding.

**Review by LinMoskovitch — APPROVED** [review]
LGTM

**Pull request merged**